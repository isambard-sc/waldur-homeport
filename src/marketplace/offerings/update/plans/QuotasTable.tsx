import { FC } from 'react';
import { Field } from 'redux-form';

import { translate } from '@waldur/i18n';
import {
  validateNonNegative,
  parseIntField,
  formatIntField,
} from '@waldur/marketplace/common/utils';
import { OfferingComponent } from '@waldur/marketplace/types';

interface QuotasTableProps {
  components: OfferingComponent[];
}

export const QuotasTable: FC<QuotasTableProps> = (props) => (
  <table className="table table-borderless">
    <thead>
      <tr>
        <th>{translate('Name')}</th>
        <th>{translate('Amount')}</th>
        <th>{translate('Units')}</th>
      </tr>
    </thead>
    <tbody>
      {props.components.map((component: OfferingComponent, index) => (
        <tr key={index}>
          <td>
            <div className="form-control-static">{component.name}</div>
          </td>
          <td>
            <Field
              component="input"
              min={0}
              className="form-control"
              name={`quotas.${component.type}`}
              type="number"
              validate={validateNonNegative}
              inputMode="numeric"
              parse={parseIntField}
              format={formatIntField}
            />
          </td>
          <td>
            <div className="form-control-static">{component.measured_unit}</div>
          </td>
        </tr>
      ))}
    </tbody>
  </table>
);
