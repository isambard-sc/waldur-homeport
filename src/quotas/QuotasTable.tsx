import { Col, Row } from 'react-bootstrap';

import { translate } from '@waldur/i18n';

import { formatQuotaValue, formatQuotaName } from './utils';

interface Quota {
  name: string;
  usage: number;
  limit: number;
}

interface Resource {
  quotas: Quota[];
}

export const QuotasTable = ({ resource }: { resource: Resource }) =>
  resource.quotas.length === 0 ? (
    <div className="row text-center">
      {translate('You have no quotas yet.')}
    </div>
  ) : (
    <div className="provider-quotas">
      {resource.quotas
        .sort((q1, q2) => q1.name.localeCompare(q2.name))
        .map((quota) => (
          <Row key={quota.name} className="mb-2">
            <Col sm={4} xs={6}>
              {formatQuotaName(quota.name)}:
            </Col>
            <Col sm={8} xs={6}>
              {translate('{usage} of {limit}', {
                usage: formatQuotaValue(quota.usage, quota.name),
                limit: formatQuotaValue(quota.limit, quota.name),
              })}
            </Col>
          </Row>
        ))}
    </div>
  );
