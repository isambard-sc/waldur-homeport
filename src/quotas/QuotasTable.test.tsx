import { render, screen } from '@testing-library/react';
import { describe, it, expect, vi } from 'vitest';

import { QuotasTable } from './QuotasTable';

vi.mock('@waldur/store/hooks', () => ({
  useTheme: () => 'light',
}));

vi.mock('@waldur/core/EChart', () => ({
  EChart: ({ options }) => (
    <div data-testid="echart">{JSON.stringify(options)}</div>
  ),
}));

const renderQuotas = (quotas) => render(<QuotasTable resource={{ quotas }} />);

describe('QuotasTable', () => {
  it('should render empty state message when no quotas', () => {
    renderQuotas([]);
    expect(screen.getByText('You have no quotas yet.')).toBeInTheDocument();
  });

  it('should render categorized quotas', () => {
    const quotas = [
      {
        name: 'ram',
        usage: 1024,
        limit: 2048,
      },
      {
        name: 'vcpu',
        usage: 2,
        limit: 4,
      },
    ];

    renderQuotas(quotas);
    expect(screen.getByText('Compute')).toBeInTheDocument();
    expect(screen.getByText('RAM')).toBeInTheDocument();
    expect(screen.getByText('vCPU count')).toBeInTheDocument();
  });

  it('should render additional quotas section for uncategorized quotas', () => {
    const quotas = [
      {
        name: 'custom_quota',
        usage: 5,
        limit: 10,
      },
    ];

    renderQuotas(quotas);
    expect(screen.getByText('Additional quotas')).toBeInTheDocument();
    expect(screen.getByText('Custom quota:')).toBeInTheDocument();
  });

  it('should handle infinite limits', () => {
    const quotas = [
      {
        name: 'unlimited_quota',
        usage: 5,
        limit: -1,
      },
    ];

    renderQuotas(quotas);
    expect(screen.getByText('5 of âˆž')).toBeInTheDocument();
  });

  it('should match quotas using regex patterns', () => {
    const quotas = [
      {
        name: 'gigabytes_custom',
        usage: 5,
        limit: 10,
      },
    ];
    renderQuotas(quotas);
    expect(screen.getByText('Storage')).toBeInTheDocument(); // Should be under Storage category
  });

  it('should sort quotas alphabetically within categories', () => {
    const quotas = [
      { name: 'vcpu', usage: 2, limit: 4 },
      { name: 'ram', usage: 1024, limit: 2048 },
    ];
    renderQuotas(quotas);
    const items = screen.getAllByText(/RAM|vCPU count/);
    expect(items[0]).toHaveTextContent('RAM');
    expect(items[1]).toHaveTextContent('vCPU count');
  });
});
