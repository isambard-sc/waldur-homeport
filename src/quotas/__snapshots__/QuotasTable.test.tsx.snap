// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`QuotasTable > renders table for quotas 1`] = `
<div>
  <div
    class="provider-quotas"
  >
    <div
      class="my-6"
    >
      <h5
        class="fw-bold mb-6"
      >
        Compute
      </h5>
      <div
        class="row"
      >
        <div
          class="col-lg-3 col-md-4 col-6"
        >
          <div
            class="resource-quota-item fw-bold d-flex mb-6"
          >
            <div
              class="content-center-center"
              style="width: 65px; height: 65px;"
            >
              <h1
                class="text-center"
              >
                <svg
                  class="animation-spin "
                  fill="currentColor"
                  height="1em"
                  viewBox="0 0 256 256"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M136,32V64a8,8,0,0,1-16,0V32a8,8,0,0,1,16,0Zm37.25,58.75a8,8,0,0,0,5.66-2.35l22.63-22.62a8,8,0,0,0-11.32-11.32L167.6,77.09a8,8,0,0,0,5.65,13.66ZM224,120H192a8,8,0,0,0,0,16h32a8,8,0,0,0,0-16Zm-45.09,47.6a8,8,0,0,0-11.31,11.31l22.62,22.63a8,8,0,0,0,11.32-11.32ZM128,184a8,8,0,0,0-8,8v32a8,8,0,0,0,16,0V192A8,8,0,0,0,128,184ZM77.09,167.6,54.46,190.22a8,8,0,0,0,11.32,11.32L88.4,178.91A8,8,0,0,0,77.09,167.6ZM72,128a8,8,0,0,0-8-8H32a8,8,0,0,0,0,16H64A8,8,0,0,0,72,128ZM65.78,54.46A8,8,0,0,0,54.46,65.78L77.09,88.4A8,8,0,0,0,88.4,77.09Z"
                  />
                </svg>
              </h1>
              <div
                class="hidden"
                style="width: 65px; height: 65px;"
              />
            </div>
            <div
              class="ms-3"
            >
              <span
                class="d-block"
              >
                <span
                  class="fs-4"
                >
                  5
                </span>
                <span
                  class="fs-7"
                >
                  /
                  30
                </span>
              </span>
              <span
                class="d-block fs-7"
              >
                Instances count
              </span>
            </div>
          </div>
        </div>
        <div
          class="col-lg-3 col-md-4 col-6"
        >
          <div
            class="resource-quota-item fw-bold d-flex mb-6"
          >
            <div
              class="content-center-center"
              style="width: 65px; height: 65px;"
            >
              <h1
                class="text-center"
              >
                <svg
                  class="animation-spin "
                  fill="currentColor"
                  height="1em"
                  viewBox="0 0 256 256"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M136,32V64a8,8,0,0,1-16,0V32a8,8,0,0,1,16,0Zm37.25,58.75a8,8,0,0,0,5.66-2.35l22.63-22.62a8,8,0,0,0-11.32-11.32L167.6,77.09a8,8,0,0,0,5.65,13.66ZM224,120H192a8,8,0,0,0,0,16h32a8,8,0,0,0,0-16Zm-45.09,47.6a8,8,0,0,0-11.31,11.31l22.62,22.63a8,8,0,0,0,11.32-11.32ZM128,184a8,8,0,0,0-8,8v32a8,8,0,0,0,16,0V192A8,8,0,0,0,128,184ZM77.09,167.6,54.46,190.22a8,8,0,0,0,11.32,11.32L88.4,178.91A8,8,0,0,0,77.09,167.6ZM72,128a8,8,0,0,0-8-8H32a8,8,0,0,0,0,16H64A8,8,0,0,0,72,128ZM65.78,54.46A8,8,0,0,0,54.46,65.78L77.09,88.4A8,8,0,0,0,88.4,77.09Z"
                  />
                </svg>
              </h1>
              <div
                class="hidden"
                style="width: 65px; height: 65px;"
              />
            </div>
            <div
              class="ms-3"
            >
              <span
                class="d-block"
              >
                <span
                  class="fs-4"
                >
                  5
                </span>
                <span
                  class="fs-7"
                >
                  /
                  128 GB
                </span>
              </span>
              <span
                class="d-block fs-7"
              >
                RAM
              </span>
            </div>
          </div>
        </div>
        <div
          class="col-lg-3 col-md-4 col-6"
        >
          <div
            class="resource-quota-item fw-bold d-flex mb-6"
          >
            <div
              class="content-center-center"
              style="width: 65px; height: 65px;"
            >
              <h1
                class="text-center"
              >
                <svg
                  class="animation-spin "
                  fill="currentColor"
                  height="1em"
                  viewBox="0 0 256 256"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M136,32V64a8,8,0,0,1-16,0V32a8,8,0,0,1,16,0Zm37.25,58.75a8,8,0,0,0,5.66-2.35l22.63-22.62a8,8,0,0,0-11.32-11.32L167.6,77.09a8,8,0,0,0,5.65,13.66ZM224,120H192a8,8,0,0,0,0,16h32a8,8,0,0,0,0-16Zm-45.09,47.6a8,8,0,0,0-11.31,11.31l22.62,22.63a8,8,0,0,0,11.32-11.32ZM128,184a8,8,0,0,0-8,8v32a8,8,0,0,0,16,0V192A8,8,0,0,0,128,184ZM77.09,167.6,54.46,190.22a8,8,0,0,0,11.32,11.32L88.4,178.91A8,8,0,0,0,77.09,167.6ZM72,128a8,8,0,0,0-8-8H32a8,8,0,0,0,0,16H64A8,8,0,0,0,72,128ZM65.78,54.46A8,8,0,0,0,54.46,65.78L77.09,88.4A8,8,0,0,0,88.4,77.09Z"
                  />
                </svg>
              </h1>
              <div
                class="hidden"
                style="width: 65px; height: 65px;"
              />
            </div>
            <div
              class="ms-3"
            >
              <span
                class="d-block"
              >
                <span
                  class="fs-4"
                >
                  3
                </span>
                <span
                  class="fs-7"
                >
                  /
                  64
                </span>
              </span>
              <span
                class="d-block fs-7"
              >
                vCPU count
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="my-6"
    >
      <h5
        class="fw-bold mb-6"
      >
        Network
      </h5>
      <div
        class="row"
      >
        <div
          class="col-lg-3 col-md-4 col-6"
        >
          <div
            class="resource-quota-item fw-bold d-flex mb-6"
          >
            <div
              class="content-center-center"
              style="width: 65px; height: 65px;"
            >
              <h1
                class="text-center"
              >
                <svg
                  class="animation-spin "
                  fill="currentColor"
                  height="1em"
                  viewBox="0 0 256 256"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M136,32V64a8,8,0,0,1-16,0V32a8,8,0,0,1,16,0Zm37.25,58.75a8,8,0,0,0,5.66-2.35l22.63-22.62a8,8,0,0,0-11.32-11.32L167.6,77.09a8,8,0,0,0,5.65,13.66ZM224,120H192a8,8,0,0,0,0,16h32a8,8,0,0,0,0-16Zm-45.09,47.6a8,8,0,0,0-11.31,11.31l22.62,22.63a8,8,0,0,0,11.32-11.32ZM128,184a8,8,0,0,0-8,8v32a8,8,0,0,0,16,0V192A8,8,0,0,0,128,184ZM77.09,167.6,54.46,190.22a8,8,0,0,0,11.32,11.32L88.4,178.91A8,8,0,0,0,77.09,167.6ZM72,128a8,8,0,0,0-8-8H32a8,8,0,0,0,0,16H64A8,8,0,0,0,72,128ZM65.78,54.46A8,8,0,0,0,54.46,65.78L77.09,88.4A8,8,0,0,0,88.4,77.09Z"
                  />
                </svg>
              </h1>
              <div
                class="hidden"
                style="width: 65px; height: 65px;"
              />
            </div>
            <div
              class="ms-3"
            >
              <span
                class="d-block"
              >
                <span
                  class="fs-4"
                >
                  6
                </span>
                <span
                  class="fs-7"
                >
                  /
                  50
                </span>
              </span>
              <span
                class="d-block fs-7"
              >
                Floating IP count
              </span>
            </div>
          </div>
        </div>
        <div
          class="col-lg-3 col-md-4 col-6"
        >
          <div
            class="resource-quota-item fw-bold d-flex mb-6"
          >
            <div
              class="content-center-center"
              style="width: 65px; height: 65px;"
            >
              <h1
                class="text-center"
              >
                <svg
                  class="animation-spin "
                  fill="currentColor"
                  height="1em"
                  viewBox="0 0 256 256"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M136,32V64a8,8,0,0,1-16,0V32a8,8,0,0,1,16,0Zm37.25,58.75a8,8,0,0,0,5.66-2.35l22.63-22.62a8,8,0,0,0-11.32-11.32L167.6,77.09a8,8,0,0,0,5.65,13.66ZM224,120H192a8,8,0,0,0,0,16h32a8,8,0,0,0,0-16Zm-45.09,47.6a8,8,0,0,0-11.31,11.31l22.62,22.63a8,8,0,0,0,11.32-11.32ZM128,184a8,8,0,0,0-8,8v32a8,8,0,0,0,16,0V192A8,8,0,0,0,128,184ZM77.09,167.6,54.46,190.22a8,8,0,0,0,11.32,11.32L88.4,178.91A8,8,0,0,0,77.09,167.6ZM72,128a8,8,0,0,0-8-8H32a8,8,0,0,0,0,16H64A8,8,0,0,0,72,128ZM65.78,54.46A8,8,0,0,0,54.46,65.78L77.09,88.4A8,8,0,0,0,88.4,77.09Z"
                  />
                </svg>
              </h1>
              <div
                class="hidden"
                style="width: 65px; height: 65px;"
              />
            </div>
            <div
              class="ms-3"
            >
              <span
                class="d-block"
              >
                <span
                  class="fs-4"
                >
                  1
                </span>
                <span
                  class="fs-7"
                >
                  /
                  100
                </span>
              </span>
              <span
                class="d-block fs-7"
              >
                Network count
              </span>
            </div>
          </div>
        </div>
        <div
          class="col-lg-3 col-md-4 col-6"
        >
          <div
            class="resource-quota-item fw-bold d-flex mb-6"
          >
            <div
              class="content-center-center"
              style="width: 65px; height: 65px;"
            >
              <h1
                class="text-center"
              >
                <svg
                  class="animation-spin "
                  fill="currentColor"
                  height="1em"
                  viewBox="0 0 256 256"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M136,32V64a8,8,0,0,1-16,0V32a8,8,0,0,1,16,0Zm37.25,58.75a8,8,0,0,0,5.66-2.35l22.63-22.62a8,8,0,0,0-11.32-11.32L167.6,77.09a8,8,0,0,0,5.65,13.66ZM224,120H192a8,8,0,0,0,0,16h32a8,8,0,0,0,0-16Zm-45.09,47.6a8,8,0,0,0-11.31,11.31l22.62,22.63a8,8,0,0,0,11.32-11.32ZM128,184a8,8,0,0,0-8,8v32a8,8,0,0,0,16,0V192A8,8,0,0,0,128,184ZM77.09,167.6,54.46,190.22a8,8,0,0,0,11.32,11.32L88.4,178.91A8,8,0,0,0,77.09,167.6ZM72,128a8,8,0,0,0-8-8H32a8,8,0,0,0,0,16H64A8,8,0,0,0,72,128ZM65.78,54.46A8,8,0,0,0,54.46,65.78L77.09,88.4A8,8,0,0,0,88.4,77.09Z"
                  />
                </svg>
              </h1>
              <div
                class="hidden"
                style="width: 65px; height: 65px;"
              />
            </div>
            <div
              class="ms-3"
            >
              <span
                class="d-block"
              >
                <span
                  class="fs-4"
                >
                  8
                </span>
                <span
                  class="fs-7"
                >
                  /
                  100
                </span>
              </span>
              <span
                class="d-block fs-7"
              >
                Security group count
              </span>
            </div>
          </div>
        </div>
        <div
          class="col-lg-3 col-md-4 col-6"
        >
          <div
            class="resource-quota-item fw-bold d-flex mb-6"
          >
            <div
              class="content-center-center"
              style="width: 65px; height: 65px;"
            >
              <h1
                class="text-center"
              >
                <svg
                  class="animation-spin "
                  fill="currentColor"
                  height="1em"
                  viewBox="0 0 256 256"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M136,32V64a8,8,0,0,1-16,0V32a8,8,0,0,1,16,0Zm37.25,58.75a8,8,0,0,0,5.66-2.35l22.63-22.62a8,8,0,0,0-11.32-11.32L167.6,77.09a8,8,0,0,0,5.65,13.66ZM224,120H192a8,8,0,0,0,0,16h32a8,8,0,0,0,0-16Zm-45.09,47.6a8,8,0,0,0-11.31,11.31l22.62,22.63a8,8,0,0,0,11.32-11.32ZM128,184a8,8,0,0,0-8,8v32a8,8,0,0,0,16,0V192A8,8,0,0,0,128,184ZM77.09,167.6,54.46,190.22a8,8,0,0,0,11.32,11.32L88.4,178.91A8,8,0,0,0,77.09,167.6ZM72,128a8,8,0,0,0-8-8H32a8,8,0,0,0,0,16H64A8,8,0,0,0,72,128ZM65.78,54.46A8,8,0,0,0,54.46,65.78L77.09,88.4A8,8,0,0,0,88.4,77.09Z"
                  />
                </svg>
              </h1>
              <div
                class="hidden"
                style="width: 65px; height: 65px;"
              />
            </div>
            <div
              class="ms-3"
            >
              <span
                class="d-block"
              >
                <span
                  class="fs-4"
                >
                  14
                </span>
                <span
                  class="fs-7"
                >
                  /
                  100
                </span>
              </span>
              <span
                class="d-block fs-7"
              >
                Security group rule count
              </span>
            </div>
          </div>
        </div>
        <div
          class="col-lg-3 col-md-4 col-6"
        >
          <div
            class="resource-quota-item fw-bold d-flex mb-6"
          >
            <div
              class="content-center-center"
              style="width: 65px; height: 65px;"
            >
              <h1
                class="text-center"
              >
                <svg
                  class="animation-spin "
                  fill="currentColor"
                  height="1em"
                  viewBox="0 0 256 256"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M136,32V64a8,8,0,0,1-16,0V32a8,8,0,0,1,16,0Zm37.25,58.75a8,8,0,0,0,5.66-2.35l22.63-22.62a8,8,0,0,0-11.32-11.32L167.6,77.09a8,8,0,0,0,5.65,13.66ZM224,120H192a8,8,0,0,0,0,16h32a8,8,0,0,0,0-16Zm-45.09,47.6a8,8,0,0,0-11.31,11.31l22.62,22.63a8,8,0,0,0,11.32-11.32ZM128,184a8,8,0,0,0-8,8v32a8,8,0,0,0,16,0V192A8,8,0,0,0,128,184ZM77.09,167.6,54.46,190.22a8,8,0,0,0,11.32,11.32L88.4,178.91A8,8,0,0,0,77.09,167.6ZM72,128a8,8,0,0,0-8-8H32a8,8,0,0,0,0,16H64A8,8,0,0,0,72,128ZM65.78,54.46A8,8,0,0,0,54.46,65.78L77.09,88.4A8,8,0,0,0,88.4,77.09Z"
                  />
                </svg>
              </h1>
              <div
                class="hidden"
                style="width: 65px; height: 65px;"
              />
            </div>
            <div
              class="ms-3"
            >
              <span
                class="d-block"
              >
                <span
                  class="fs-4"
                >
                  1
                </span>
                <span
                  class="fs-7"
                >
                  /
                  100
                </span>
              </span>
              <span
                class="d-block fs-7"
              >
                Subnet count
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="my-6"
    >
      <h5
        class="fw-bold mb-6"
      >
        Storage
      </h5>
      <div
        class="row"
      >
        <div
          class="col-lg-3 col-md-4 col-6"
        >
          <div
            class="resource-quota-item fw-bold d-flex mb-6"
          >
            <div
              class="content-center-center"
              style="width: 65px; height: 65px;"
            >
              <h1
                class="text-center"
              >
                <svg
                  class="animation-spin "
                  fill="currentColor"
                  height="1em"
                  viewBox="0 0 256 256"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M136,32V64a8,8,0,0,1-16,0V32a8,8,0,0,1,16,0Zm37.25,58.75a8,8,0,0,0,5.66-2.35l22.63-22.62a8,8,0,0,0-11.32-11.32L167.6,77.09a8,8,0,0,0,5.65,13.66ZM224,120H192a8,8,0,0,0,0,16h32a8,8,0,0,0,0-16Zm-45.09,47.6a8,8,0,0,0-11.31,11.31l22.62,22.63a8,8,0,0,0,11.32-11.32ZM128,184a8,8,0,0,0-8,8v32a8,8,0,0,0,16,0V192A8,8,0,0,0,128,184ZM77.09,167.6,54.46,190.22a8,8,0,0,0,11.32,11.32L88.4,178.91A8,8,0,0,0,77.09,167.6ZM72,128a8,8,0,0,0-8-8H32a8,8,0,0,0,0,16H64A8,8,0,0,0,72,128ZM65.78,54.46A8,8,0,0,0,54.46,65.78L77.09,88.4A8,8,0,0,0,88.4,77.09Z"
                  />
                </svg>
              </h1>
              <div
                class="hidden"
                style="width: 65px; height: 65px;"
              />
            </div>
            <div
              class="ms-3"
            >
              <span
                class="d-block"
              >
                <span
                  class="fs-4"
                >
                  2
                </span>
                <span
                  class="fs-7"
                >
                  /
                  50
                </span>
              </span>
              <span
                class="d-block fs-7"
              >
                Snapshots count
              </span>
            </div>
          </div>
        </div>
        <div
          class="col-lg-3 col-md-4 col-6"
        >
          <div
            class="resource-quota-item fw-bold d-flex mb-6"
          >
            <div
              class="symbol symbol-65px symbol-circle"
            >
              <div
                class="image-placeholder"
                style="width: 65px; height: 65px; background-color: transparent;"
              >
                <div
                  class="symbol-label fs-1 fw-bolder text-muted w-100 h-100"
                >
                  50
                </div>
              </div>
            </div>
            <div
              class="ms-3"
            >
              <span
                class="d-block"
              >
                <span
                  class="fs-4"
                >
                  50
                </span>
                <span
                  class="fs-7"
                >
                  /
                  ∞
                </span>
              </span>
              <span
                class="d-block fs-7"
              >
                Snapshots size
              </span>
            </div>
          </div>
        </div>
        <div
          class="col-lg-3 col-md-4 col-6"
        >
          <div
            class="resource-quota-item fw-bold d-flex mb-6"
          >
            <div
              class="content-center-center"
              style="width: 65px; height: 65px;"
            >
              <h1
                class="text-center"
              >
                <svg
                  class="animation-spin "
                  fill="currentColor"
                  height="1em"
                  viewBox="0 0 256 256"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M136,32V64a8,8,0,0,1-16,0V32a8,8,0,0,1,16,0Zm37.25,58.75a8,8,0,0,0,5.66-2.35l22.63-22.62a8,8,0,0,0-11.32-11.32L167.6,77.09a8,8,0,0,0,5.65,13.66ZM224,120H192a8,8,0,0,0,0,16h32a8,8,0,0,0,0-16Zm-45.09,47.6a8,8,0,0,0-11.31,11.31l22.62,22.63a8,8,0,0,0,11.32-11.32ZM128,184a8,8,0,0,0-8,8v32a8,8,0,0,0,16,0V192A8,8,0,0,0,128,184ZM77.09,167.6,54.46,190.22a8,8,0,0,0,11.32,11.32L88.4,178.91A8,8,0,0,0,77.09,167.6ZM72,128a8,8,0,0,0-8-8H32a8,8,0,0,0,0,16H64A8,8,0,0,0,72,128ZM65.78,54.46A8,8,0,0,0,54.46,65.78L77.09,88.4A8,8,0,0,0,88.4,77.09Z"
                  />
                </svg>
              </h1>
              <div
                class="hidden"
                style="width: 65px; height: 65px;"
              />
            </div>
            <div
              class="ms-3"
            >
              <span
                class="d-block"
              >
                <span
                  class="fs-4"
                >
                  295
                </span>
                <span
                  class="fs-7"
                >
                  /
                  295 GB
                </span>
              </span>
              <span
                class="d-block fs-7"
              >
                Storage
              </span>
            </div>
          </div>
        </div>
        <div
          class="col-lg-3 col-md-4 col-6"
        >
          <div
            class="resource-quota-item fw-bold d-flex mb-6"
          >
            <div
              class="content-center-center"
              style="width: 65px; height: 65px;"
            >
              <h1
                class="text-center"
              >
                <svg
                  class="animation-spin "
                  fill="currentColor"
                  height="1em"
                  viewBox="0 0 256 256"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M136,32V64a8,8,0,0,1-16,0V32a8,8,0,0,1,16,0Zm37.25,58.75a8,8,0,0,0,5.66-2.35l22.63-22.62a8,8,0,0,0-11.32-11.32L167.6,77.09a8,8,0,0,0,5.65,13.66ZM224,120H192a8,8,0,0,0,0,16h32a8,8,0,0,0,0-16Zm-45.09,47.6a8,8,0,0,0-11.31,11.31l22.62,22.63a8,8,0,0,0,11.32-11.32ZM128,184a8,8,0,0,0-8,8v32a8,8,0,0,0,16,0V192A8,8,0,0,0,128,184ZM77.09,167.6,54.46,190.22a8,8,0,0,0,11.32,11.32L88.4,178.91A8,8,0,0,0,77.09,167.6ZM72,128a8,8,0,0,0-8-8H32a8,8,0,0,0,0,16H64A8,8,0,0,0,72,128ZM65.78,54.46A8,8,0,0,0,54.46,65.78L77.09,88.4A8,8,0,0,0,88.4,77.09Z"
                  />
                </svg>
              </h1>
              <div
                class="hidden"
                style="width: 65px; height: 65px;"
              />
            </div>
            <div
              class="ms-3"
            >
              <span
                class="d-block"
              >
                <span
                  class="fs-4"
                >
                  8
                </span>
                <span
                  class="fs-7"
                >
                  /
                  50
                </span>
              </span>
              <span
                class="d-block fs-7"
              >
                Volumes count
              </span>
            </div>
          </div>
        </div>
        <div
          class="col-lg-3 col-md-4 col-6"
        >
          <div
            class="resource-quota-item fw-bold d-flex mb-6"
          >
            <div
              class="symbol symbol-65px symbol-circle"
            >
              <div
                class="image-placeholder"
                style="width: 65px; height: 65px; background-color: transparent;"
              >
                <div
                  class="symbol-label fs-1 fw-bolder text-muted w-100 h-100"
                >
                  245
                </div>
              </div>
            </div>
            <div
              class="ms-3"
            >
              <span
                class="d-block"
              >
                <span
                  class="fs-4"
                >
                  245
                </span>
                <span
                  class="fs-7"
                >
                  /
                  ∞
                </span>
              </span>
              <span
                class="d-block fs-7"
              >
                Volumes size
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
